src/agent/rustjail/src/cgroups/systemd/subsystem/cpu.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021-2022 Kata Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::super::common::{CgroupHierarchy, Properties};
+use super::transformer::Transformer;
+
+use anyhow::Result;
+use oci::{LinuxCpu, LinuxResources};
+use zbus::zvariant::Value;
+
+const BASIC_SYSTEMD_VERSION: &str = "242";
+const DEFAULT_CPUQUOTAPERIOD: u64 = 100 * 1000;
+const SEC2MICROSEC: u64 = 1000 * 1000;
+const BASIC_INTERVAL: u64 = 10 * 1000;
+
+pub struct Cpu {}
+
+impl Transformer for Cpu {
+    fn apply(
+        r: &LinuxResources,
+        properties: &mut Properties,
+        cgroup_hierarchy: &CgroupHierarchy,
+        systemd_version: &str,
+    ) -> Result<()> {
+        if let Some(cpu_resources) = &r.cpu {
+            match cgroup_hierarchy {
+                CgroupHierarchy::Legacy => Self::legacy_apply(cpu_resources, properties)?,
+                CgroupHierarchy::Unified => {
+                    Self::unified_apply(cpu_resources, properties, systemd_version)?
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Cpu {
+    // v1:
+    // cpu.shares <-> CPUShares
+    fn legacy_apply(cpu_resources: &LinuxCpu, properties: &mut Properties) -> Result<()> {
```

Comment:
Let's just call it `apply_v1` and `apply_v2`.

Reply:
If we're changing it, please, let's keep it consistent and change `legacy` -> `v1`, and `unified` -> `v2` everywhere.